src/main/java/Duke.java

```diff
@@ -1,10 +1,108 @@
+        while (!isClosed) {
+            try {
+                String command = sc.nextLine();
+                if (command.equals("bye")) {
+                    bye();
+                } else if (command.equals("list")) {
+                    printList();
+                } else {
+                    if (command.startsWith("mark")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("mark");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.markAsDone(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("mark");
+                        }
+                    } else if (command.startsWith("unmark")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("unmark");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.markAsNotDone(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("unmark");
+                        }
+                    } else if (command.startsWith("todo")) {
+                        tasks.addTodo(command);
+                    } else if (command.startsWith("deadline")) {
+                        tasks.addDeadline(command);
+                    } else if (command.startsWith("event")) {
+                        tasks.addEvent(command);
+                    } else if (command.startsWith("delete")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("delete");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.deleteTask(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("delete");
+                        }
+                    } else {
+                        throw new DukeException("You have entered an invalid command");
+                    }
+                }
+            } catch (DukeException ex) {
+                drawLine();
+                indentMessage(ex.getMessage());
+                drawLine();
+            }
+        }
+    }
+
+    private static void greetingMessage() {
```

Comment:
Would using a verb for the method name would be more suitable?

Reply:
Right, I'll take it into consideration. Thank you for your comment.